clang/lib/Driver/SanitizerArgs.cpp

```diff
@@ -1159,7 +1159,8 @@ SanitizerArgs::SanitizerArgs(const ToolChain &TC,
 
   LinkRuntimes =
       Args.hasFlag(options::OPT_fsanitize_link_runtime,
-                   options::OPT_fno_sanitize_link_runtime, LinkRuntimes);
+                   options::OPT_fno_sanitize_link_runtime, LinkRuntimes) &&
```

Comment:
Maybe `-r` should control the default? That would let you do `-fsanitize-link-runtime -r` if you really wanted to link the runtime into an object file for some reason.

Reply:
I guess that linking sanitizer runtimes into object file doesnt make sense but okay :)

Reply:
The third argument to `hasFlag` is the default, you could pass in the `-r` value there. `LinkRuntimes` is only set here so you don't need to care about the previous value.

Reply:
Done

Reply:
No I mean
```suggestion
                   options::OPT_fno_sanitize_link_runtime, !Args.hasArg(options::OPT_r));
```

Reply:
done